types/mydecimal.go

```diff
@@ -1301,6 +1305,12 @@ func (d *MyDecimal) ToHashKey() ([]byte, error) {
 		// thus ErrTruncated may be raised, we can ignore it here.
 		err = nil
 	}
+	precBuff := make([]byte, 4)
+	digitBuff := make([]byte, 4)
+	binary.LittleEndian.PutUint32(precBuff, uint32(prec))
+	binary.LittleEndian.PutUint32(digitBuff, uint32(digitsFrac))
```

Comment:
I think storing `digitsFrac` is enough after removing the leading and trailing zeros.